### DISCUSSION TITLE
Custom GPTs Programmatic Access

### DISCUSSION THREAD

**Discussion by stoufa** [openai/openai-python#1832] [Q&A]
Is it possible to programmatically use [Custom GPTs](https://www.youtube.com/watch?v=ABVwhZWg1Uk) with this library?

**Comment by kalyanamdewri** [comment]
Not directly as a Custom GPT object or GPT ID through `openai-python`.

Custom GPTs are configured and used inside ChatGPT. The API path is separate: build the same behavior in your app with the OpenAI API by carrying over the GPT configuration into API primitives:

- GPT instructions -> `instructions` / developer message in your API call
- GPT knowledge files -> file search / vector stores
- GPT actions -> function calling or your own API/tool layer
- GPT web capability -> the API web search tool when that is appropriate
- GPT code/data analysis behavior -> the API code interpreter tool when you need hosted Python execution

The relevant distinction in the OpenAI docs is that GPTs are no-code assistants built and used inside ChatGPT, while API assistants/integrations are developer-built experiences inside your own app. The GPTs FAQ also says GPTs are not the way to embed ChatGPT in an external product; for that, use the API.

Docs:
- https://help.openai.com/en/articles/8554407-gpts-in-chatgpt
- https://developers.openai.com/api/docs/guides/tools

So if you already have a Custom GPT and want programmatic access, the practical route is to recreate its instructions, files, and tool behavior in an API-backed assistant/agent rather than trying to call the ChatGPT GPT directly.
